### PR TITLE
Don't clear editor contents when writing to clipboard fails

### PR DESCRIPTION
**Pull request by swazrgb**
Instead present behavior string via a prompt dialogue where the user can copy it

### CODE REVIEW

File: website/index.html [hunk 1]
```diff
@@ -190,18 +190,11 @@
...
-        copied = true;
         alert("Copied to clipboard");
-        } catch {}
-      } 
-      if (!copied) {
-        txtModel.setValue(text);
-        editor.setModel(txtModel);
-        editor.setSelection(txtModel.getFullModelRange());
+      } catch {
+        prompt("Behavior string:", text);
```
Comment (ribrdb) [inline website/index.html:197]: How does this look if the string is really long? Is it still copyable?

Comment (swazrgb) [reply website/index.html:197]: The prompt looks like this:

image

In the screenshot I generated a ~6mb string and displayed it using `prompt("Behavior string":, text)`. The string can be fully copied from the prompt dialog.

**Comment by swazrgb** [comment]
Due to the async usage in the onclick method the user may interact with the page prior to the async block finishing, which has the effect of the browser not allowing us to set the clipboard.

Previously the text input would then be replaced with the behavior string. As a new user, this confused me a lot since it felt like I'd lose the behavior typescript I was working on.

My other PR has reduced the async in the onclick method which reduces the chance of this happening, but it can still happen. Personally I vastly prefer the prompt in case writing to the clipboard fails.

Somewhat related, in a future PR I'd like to move the compilation to the background, so we try to compile periodically as the user is typing, so the errors can be displayed inline as they are working on the behavior.

A benefit of doing this is that the export button can become non-async, simply copying the behavior string of the already compiled behavior to the clipboard.

**Comment by ribrdb** [comment]
> Somewhat related, in a future PR I'd like to move the compilation to the background, so we try to compile periodically as the user is typing, so the errors can be displayed inline as they are working on the behavior.

Good idea!

**Review by ribrdb — APPROVED** [review]

**Pull request merged**